src/Utils/Rotation.cpp

```
@@ -312,15 +312,335 @@ void utils::Rotation::checkUnitary()
+                            {
+                                Um(0, dim_a) *= -1.0;
+                                Um(1, dim_a) *= -1.0;
+                                Um(2, dim_a) *= -1.0;
+                            }
+                            done = 1;
+                            break; // out of for
+                        }
+                    }
+                }
+
+                if ((!done) && (modified_SVD == 1))
+                {
+                    // Handle situation:
+                    // negative determinant (Tet is inverted in solid mechanics)
+                    //    - check if det(U) == -1
+                    //    - If yes, then negate the minimal element of Sigma
+                    //      and the corresponding column of U
+
+                    float det_U = Um.determinant();
+                    if (det_U < 0.0)
+                    {
+                        // negative determinant
+                        // find the smallest singular value (they are all non-negative)
+                        int smallest_singular_value_idx = 0;
+                        for(int dim=1; dim<3; dim++)
+                            if (lambda(dim) < lambda(smallest_singular_value_idx))
+                                smallest_singular_value_idx = dim;
+
+                        // negate the smallest singular value
+                        lambda(smallest_singular_value_idx) *= -1.0;
+                        Um(0, smallest_singular_value_idx) *= -1.0;
+                        Um(1, smallest_singular_value_idx) *= -1.0;
+                        Um(2, smallest_singular_value_idx) *= -1.0;
+                    }
+                }
+            }
+            R = Um * Vm.transpose();
+        }
+        else
+        {
+            // R = Mk^T
+            R = Mk.transpose();
+        }
+
+
+        std::cout << "On s'est rendu, youpi!\n";
+
+        // utils::Matrix3d mat;
 
         utils::Error::check(false,
```

That is my mistake, but since we already know that this should raise an error, `utils::Error::raise("message")` is a more intuitive method to call (instead of `check` which is more designed to validate the first argument, that is, if the first argument is false, it raises an error with the message, but if it is true, it does not)